dev/create-release/release-build.sh

```diff
@@ -111,13 +111,17 @@ fi
 # different versions of Scala are supported.
 BASE_PROFILES="-Pmesos -Pyarn"
 PUBLISH_SCALA_2_10=0
+PUBLISH_SCALA_2_12=0
 SCALA_2_10_PROFILES="-Pscala-2.10"
 SCALA_2_11_PROFILES=
-SCALA_2_12_PROFILES="-Pscala-2.12"
+SCALA_2_12_PROFILES="-Pscala-2.12 -Pkafka-0-8"
```

Comment:
@cloud-fan wait, no this won't work. You can't build Kafka 0.8 for Scala 2.12. This env var is only supposed to have the scala-2.12 profile. The logic below was correct without this change.

Reply:
So we only set the kafka 0.8 profile if and only the spark version > 2.3 and it's a scala 2.11 build?

Reply:
Yes that's correct. It shouldn't be built for Scala 2.10 or 2.12, and did not exist before 2.3.0.

Reply:
somehow it works when I run it... Anyway I'll send a followup PR to change it back.

Reply:
Hm, it shouldn't, in that there's no 2.12 artifacts for Kafka 0.8

https://search.maven.org/search?q=g:org.apache.kafka%20AND%20a:kafka_2.12&core=gav
vs
```
      <groupId>org.apache.kafka</groupId>
      <artifactId>kafka_${scala.binary.version}</artifactId>
      <version>0.8.2.1</version>
```

Maybe it actually fails at the end of the 2.12 build, after a lot of it has succeeded